src/interactive-window/editor-integration/codeLensFactory.ts

```diff
@@ -62,12 +61,9 @@ export class CodeLensFactory implements ICodeLensFactory {
         this.workspace.onDidGrantWorkspaceTrust(() => this.codeLensCache.clear(), this, disposables);
         this.configService.getSettings(undefined).onDidChange(this.onChangedSettings, this, disposables);
         notebook.onDidChangeNotebookCellExecutionState(this.onDidChangeNotebookCellExecutionState, this, disposables);
-        kernelProvider.onDidDisposeKernel(
+        kernelProvider.onDidDisposeNotebookKernel(
             (kernel) => {
-                const notebook = getAssociatedNotebookDocument(kernel);
-                if (notebook) {
-                    this.notebookData.delete(notebook.uri.toString());
-                }
+                this.notebookData.delete((kernel as INotebookKernel).notebook.uri.toString());
```

Comment:
We don't need to cast this. 
If we haev a special event just for INotebookKernel, then the argument kernel should be of type INoteobookKernel